go.mod

```diff
@@ -6,6 +6,7 @@ require (
 	github.com/MakeNowJust/heredoc v1.0.0
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/coredns/corefile-migration v1.0.12
+	github.com/coreos/ignition v0.35.0
```

Comment:
As we mainly focus on supporting Flatcar Container Linux, which right now still uses Ignition v2 and versions 0.35+ (so 2.0+) removes v2 support. This is the latest upstream version containing v2 code.

Reply:
Are there plans to eventually get rid of this dependency in exchange for a newer one?

Reply:
Hey folks, good catch. We are working on a path forward and bring it to discussion in the Wednesday (Nov 3rd) office hours.

Reply:
Notes from our discussion in the office hours on Nov 3rd:
- Flatcar team actively maintains v1 at https://github.com/flatcar-linux/ignition
- We will update the module to point to github.com/flatcar-linux/ignition/v1
- v1 implementation is not Flatcar Linux specific; any distro using v1 will be supported by this PR
- the PR is structured so that v2 support can be added in parallel to v2, if desired ([see also](https://github.com/kubernetes-sigs/cluster-api/pull/4172#issuecomment-777455472))

Reply:
I'd suggest using `v0`, if possible, to avoid making the numbering even more confusing.  Ignition 0.x supports spec 1 and 2.x, and Ignition 2.x supports spec 3.x.

Reply:
Hi there - according to the Go [documentation]():
> Major version suffixes are not allowed at major versions v0 or v1.[^1]

I'm not sure it's worth to go against the Go recommendations regarding the module versioning.

> If the module is released at major version 2 or higher, the module path must end with a major version suffix like /v2 [^2]

Based on this, I'd suggest to only rename `github.com/coreos/ignition` to `github.com/flatcar-linux/ignition` into `go.mod` - `v0/v1` is implicitly hold by this declaration.

On the other hand, a PR has been opened: https://github.com/flatcar-linux/ignition/pull/26 to bring Go module support to `flatcar-linux/ignition` - once this PR merged, we will tag a release of `flatcar-linux/ignition` in order to have a proper dependency in the `go.mod` of this PR.

Doing this does not conflict with the latest bullet:
> * the PR is structured so that v2 support can be added in parallel to v2, if desired

since this is what @bgilbert  already done in the `coreos/ign-converter`: https://github.com/coreos/ign-converter/blob/984c88c5be3220a314d9474ce61ab5bf9e60dab2/translate/v24tov31/v24tov31.go#L23-L26 - having both versions of the module in the same file.

[^1]: https://golang.org/ref/mod#major-version-suffixes
[^2]: https://golang.org/ref/mod#module-path

Reply:
We've released `github.com/flatcar-linux/ignition v0.36.1`. Waiting for https://github.com/flatcar-linux/container-linux-config-transpiler/pull/14 to be merged and released before we can replace `github.com/coreos/ignition` since `v0.9.1` still depends on `github.com/coreos/ignition`.

Reply:
Fixed. We are now consuming `github.com/flatcar-linux/ignition v0.36.1` as discussed above.

This can be resolved.

Reply:
Marking as resolved then.